src/app/components/create-retro-form/create-retro-form.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { MdDialogRef, MdSelect } from '@angular/material';
+import { Retro } from '../../models/retro';
+import { Phase } from '../../models/phase';
+import { AngularFire, FirebaseListObservable } from 'angularfire2';
+
+@Component({
+  selector: 'app-create-retro-form',
+  templateUrl: './create-retro-form.component.html',
+})
+
+export class CreateRetroFormComponent {
+  retroName: string;
+  numberOfVotes = 2;
+  phases = [];
+  af;
+
+  constructor(af: AngularFire, private router: Router, private dialogRef: MdDialogRef<CreateRetroFormComponent>)
+  {
+    this.af = af;
+    while(this.phases.length < 3){
+      this.addPhase();
+    }
+  }
+
+  addPhase(){
+    this.phases.push(new Phase("", 1, "", this.phases.length + 1));
```

Comment:
When I tried to create a retro and it routed to the next page, it threw an error saying that `currentPhaseId` doesn't exist (I think we need to update the `1` in this line to use the first phase's `$key` as the `currentPhaseId`).